src/main/java/com/rapid7/jenkinspider/PostBuildScan.java

```diff
@@ -241,6 +244,40 @@ private void SaveToFile(String filename, String data) {
             e.printStackTrace();
         }
     }
+    /**
+     *
+     * @param filename
+     * @param data
+     */
+    private void SaveToFile(String filename, InputStream inputStream) {
+        if (inputStream == null) {
+            throw new RuntimeException("unexpected invalid input stream encountered while saving file");
+        }
+
+        File file = new File(filename);
+
+        try {
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+
+            InputStream bufferedInputStream = new BufferedInputStream(inputStream); 
+            FileOutputStream outputStream = new FileOutputStream(file);
```

Comment:
You could utilise a try with resources to prevent manually calling `close` on both these streams.

Using a sample pulled from google:
```
try(InnerResource inner = new InnerResource();
    OuterResource outer = new OuterResource(inner, new RuntimeException("OUTER")))
{
   out.println(outer);
}
catch (Exception exception)
{
   out.println("ERROR: " + exception);
}
```
Creating the resources inside parentheses after the `try` enables auto closing, provided the resources implement `Closeable` interface (which they do in this case)